### PR TITLE
Shorten now supports Google's API Key and userIP request parameters.

### PR DESCRIPTION
**Pull request by shpupti**
Used to bypass Google's request rate limits.
Tests might need some refactoring but should be good enough for now.

### CODE REVIEW

**Comment by shpupti** [comment]
Can someone please review this?

**Comment by kevinchandler** [comment]
:+1:

File: lib/googl/shorten.rb [hunk 1]
```diff
@@ -8,10 +8,20 @@ class Shorten < Base
...
     #
-    def initialize(long_url)
+    def initialize(long_url, user_ip = nil, api_key = nil)
       modify_headers('Content-Type' => 'application/json')
-      options = {"longUrl" => long_url}.to_json
-      resp = post(API_URL, :body => options)
+      options = {"longUrl" => long_url}
+      shorten_url = API_URL
+
+      if (user_ip != nil && !user_ip.empty?)
```
Comment (fridgamarator) [inline lib/googl/shorten.rb]: `.blank?` accomplished both of these

**Comment by fridgamarator** [comment]
:+1:

**Comment by jtadeulopes** [comment]
@shpupti Thanks! But, can you update the readme?
https://github.com/zigotto/googl#shorten-a-long-url

File: lib/googl/shorten.rb (hunk 1 above)
Comment (jtadeulopes) [inline lib/googl/shorten.rb]: @fridgamarator see https://github.com/zigotto/googl/pull/5

Comment (fridgamarator) [inline lib/googl/shorten.rb]: ahhh, totally makes sense, sometimes I get stuck in rails mode

**Comment by shpupti** [comment]
@jtadeulopes Sure thing.

**Comment by shpupti** [comment]
Oops, apparently my MD syntax is broken.
Will fix once I find out how to properly comment in MD.

**Comment by shpupti** [comment]
Fixed, sorry for the bad commit.

**Comment by jtadeulopes** [comment]
Successfully registered gem: googl (0.7.0)

**Pull request merged**